File: 005/005.c
#include <stdio.h>

long long int gcd(long long int a, long long int b)
{
    long long int c;

    while (b) {
        c = b;
        b = a % b;
        a = c;
    }

    return a;
}

long long int lcm(long long int a, long long int b)
{
    return (a * b) / gcd(a, b);
}

long long int get_solution()
{
    long long int i, s = 1;

    for (i = 2; i < 21; ++i) {
        s = lcm(s, i);
    }
    return s;
}

int main(void)
{
    printf("Result %lld\n", get_solution());
    return 0;
}

File: 005/005.cs
using System;
using System.Linq;

public class Solution005
{
    static int gcd(int a, int b)
    {
        while (b != 0) {
            int c = b;
            b = a % b;
            a = c;
        }
        return a;
    }

    static int lcm(int a, int b)
    {
        return a * b / gcd(a, b);
    }

    static void Main()
    {
        var result = Enumerable.Range(1,20)
            .Aggregate((a,b) => lcm(a,b));
        Console.WriteLine("Result {0}\n", result);
    }
}
